src/database/schema/ChatSchema.ts

```diff
@@ -209,6 +209,32 @@ class ChatSchema implements BaseDbSchema {
     });
   }
 
+  static fromGetAllChannelAPI(channelId, json): ChatSchema {
+    let rawJson = null;
+
+    try {
+      rawJson = JSON.stringify(json);
+    } catch (e) {
+      console.log('error stringify');
+      console.log(e);
+    }
+
+    return new ChatSchema({
+      id: json?.id,
+      channelId,
+      userId: json?.user?.id,
+      message: (json?.text || json?.message) ?? '',
+      type: json?.type,
+      createdAt: json?.created_at,
+      updatedAt: json?.created_at,
+      rawJson,
+      user: null,
+      status: 'sent',
+      isMe: false,
+      isContinuous: false
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The method `fromGetAllChannelAPI` has been added. It seems to convert the JSON response from an API call into a `ChatSchema` object. However, there's no error handling for when `json` is null or doesn't have the expected structure. Consider adding checks and error handling for these cases.



<!-- This is an auto-generated comment by OSS CodeRabbit -->